llvm/lib/CodeGen/CodeGenPrepare.cpp

```diff
@@ -2634,6 +2634,15 @@ struct ExtAddrMode : public TargetLowering::AddrMode {
   void print(raw_ostream &OS) const;
   void dump() const;
 
+  void replaceWith(Value *From, Value *To) {
+    if (BaseReg == From)
```

Comment:
Are all cases below covered by the one new test?

Reply:
No. I have no idea how to generate tests that would cover all these branches :(